OSDOCS-9725 ALBO Install Doc Procedure Updates (#73583)
openshift/openshift-docs: modules/installing-aws-load-balancer-operator-cli.adoc

// Module included in the following assemblies:
//
// * networking/aws_load_balancer_operator/install-aws-load-balancer-operator.adoc

:_mod-docs-content-type: PROCEDURE
[id="nw-installing-aws-load-balancer-operator-cli_{context}"]
= Installing the AWS Load Balancer Operator with the command line

> 

Reviewer: Procedure modules should have an intro sentence. See: https://github.com/redhat-documentation/modular-docs/blob/main/modular-docs-manual/files/TEMPLATE_PROCEDURE_doing-one-procedure.adoc?plain=1#L25

.Prerequisites

* You have logged in to the {product-title} web console as a user with `cluster-admin` permissions.

* Your cluster is configured with AWS as the platform type and cloud provider.

> * You are logged into the command line client `oc`.

Reviewer: Should write oc like below. See: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/term_glossary.adoc#openshift-cli

```suggestion
* You are logged into the OpenShift CLI (`oc`).
```

Author: ty

.Procedure

. Create the `aws-load-balancer-operator` namespace:
+
[source,terminal]
----

> oc apply -f namespace.yaml

Reviewer: These commands look to be out of order. This is trying to apply a file to the cluster that hasn't been created yet. Also, the `cat` command doesn't actually do anything other than output the contents of a file that already exists.

We have a lot of examples in the repo with procedures that create a file with certain contents, and then apply the file to the cluster.

There are lots of ways to do it, but the idea should be the same that the file is created first, then applied to the cluster. 

Here are a few examples you could model after:

* https://docs.openshift.com/container-platform/4.15/security/cert_manager_operator/cert-manager-operator-issuer-acme.html#cert-manager-acme-http01_cert-manager-operator-issuer-acme
* https://docs.openshift.com/container-platform/4.15/networking/metallb/about-advertising-ipaddresspool.html
* https://docs.openshift.com/container-platform/4.15/logging/log_storage/installing-log-storage.html

So recommend looking at these and reworking these steps to follow something similar to one of these.

cat namespace.yaml
----
+
Edit the namespace to include:
+
[source,yaml]
----
apiVersion: v1
kind: Namespace
metadata:
  name: aws-load-balancer-operator
----

. Create the `credentialsrequest` file:
+

> [source, terminal]

Reviewer: No space preferred

```suggestion
[source,terminal]
```

----

> oc apply -f credentialsrequest.yaml

Reviewer: Same for all of these to only have a single command in each code block, and making sure to have a "$" prompt.

cat credentialsrequest.yaml
----
+
Edit the file to include:
+
[source,yaml]
----
apiVersion: cloudcredential.openshift.io/v1
kind: CredentialsRequest
metadata:
  name: aws-load-balancer-operator
  namespace: openshift-cloud-credential-operator
spec:
  providerSpec:
    apiVersion: cloudcredential.openshift.io/v1
    kind: AWSProviderSpec
    statementEntries:
      - action:
          - ec2:DescribeSubnets
        effect: Allow
        resource: "*"
      - action:
          - ec2:CreateTags
          - ec2:DeleteTags
        effect: Allow
        resource: arn:aws:ec2:*:*:subnet/*
      - action:
          - ec2:DescribeVpcs
        effect: Allow
        resource: "*"
  secretRef:
    name: aws-load-balancer-operator
    namespace: aws-load-balancer-operator
  serviceAccountNames:
    - aws-load-balancer-operator-controller-manager

> ----

Reviewer 2: The `CredentialsRequest` resource is only need on the version before 4.14. On OCP 4.14 `CredentialsRequest` is created by the operator.

Author: Cherrypick fix

. Create the `operatorgroup` file:
+
[source,terminal]
----
oc apply -f operatorgroup.yaml
cat operatorgroup.yaml
----
+
Edit the file to include:
+
[source,yaml]
----
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: aws-lb-operatorgroup
  namespace: aws-load-balancer-operator
spec:
  upgradeStrategy: Default
----

. Create the `subscription` file:
+
[source,terminal]
----
oc apply -f subscription.yaml
cat subscription.yaml
----
+
Edit the file to include:
+
[source,yaml]
----
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: aws-load-balancer-operator
  namespace: aws-load-balancer-operator
spec:
  channel: stable-v1
  installPlanApproval: Automatic
  name: aws-load-balancer-operator
  source: qe-app-registry
  sourceNamespace: openshift-marketplace
----

.Verification

> +

Reviewer: These "+" are rendering in the preview

![image](https://github.com/openshift/openshift-docs/assets/7443887/e30f355a-53d2-4bd9-b1e3-86bce2efc7f2)

Also, there is no step/text before the commands. If it's a single step, it should just be an unnumbered bullet (`*`). But if you make it two steps (because there are two commands), then it would be numbered (`.`).

Then, you would need these "+" in between the step and the code block to keep it together.

[source,terminal]
----

> oc -n aws-load-balancer-operator get sub aws-load-balancer-operator --template='{{.status.installplan.name}}{{"\n"}}'

Reviewer: There are two commands in this code block. There should only be a single command in each code block. See: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#single-command-per-code-block

There also needs to be a prompt in front of each terminal command, like below, for all of the oc commands.

```suggestion
$ oc -n aws-load-balancer-operator get sub aws-load-balancer-operator --template='{{.status.installplan.name}}{{"\n"}}'
```

oc -n aws-load-balancer-operator get ip <install_plan_name> --template='{{.status.phase}}{{"\n"}}'
----
+

> The output must be `Complete`. 

Reviewer: Not critical, but you could consider showing an `.Example output` code block of what the expected output for these commands are. It's not clear, since there are two commands, and I expect one of them is expected to show "Complete". But not sure if both of them are, or just one of them.

// Module included in the following assemblies:
//
// * networking/aws_load_balancer_operator/install-aws-load-balancer-operator.adoc

:_mod-docs-content-type: PROCEDURE
[id="nw-installing-aws-load-balancer-operator-cli_{context}"]
= Installing the AWS Load Balancer Operator with the command line

.Prerequisites

* You have logged in to the {product-title} web console as a user with `cluster-admin` permissions.

* Your cluster is configured with AWS as the platform type and cloud provider.

* You are logged into the command line client `oc`.

.Procedure

. Create the `aws-load-balancer-operator` namespace:

[source,terminal]
----
oc apply -f namespace.yaml
cat namespace.yaml
----

> 

Reviewer 3: ```suggestion
+
```

Edit the namespace to include:

> 

Reviewer 3: ```suggestion
+
```

[source,yaml]
----
apiVersion: v1
kind: Namespace
metadata:
  name: aws-load-balancer-operator
----

. Create the `credentialsrequest` file:

[source, terminal]
----
oc apply -f credentialsrequest.yaml
cat credentialsrequest.yaml
----

> 

Reviewer 3: ```suggestion
+
```

Edit the file to include:

> 

Reviewer 3: ```suggestion
+
```

[source,yaml]
----
apiVersion: cloudcredential.openshift.io/v1
kind: CredentialsRequest
metadata:
  name: aws-load-balancer-operator
  namespace: openshift-cloud-credential-operator
spec:
  providerSpec:
    apiVersion: cloudcredential.openshift.io/v1
    kind: AWSProviderSpec
    statementEntries:
      - action:
          - ec2:DescribeSubnets
        effect: Allow
        resource: "*"
      - action:
          - ec2:CreateTags
          - ec2:DeleteTags
        effect: Allow
        resource: arn:aws:ec2:*:*:subnet/*
      - action:
          - ec2:DescribeVpcs
        effect: Allow
        resource: "*"
  secretRef:
    name: aws-load-balancer-operator
    namespace: aws-load-balancer-operator
  serviceAccountNames:
    - aws-load-balancer-operator-controller-manager
----

. Create the `operatorgroup` file:
+
[source,terminal]
----
oc apply -f operatorgroup.yaml
cat operatorgroup.yaml
----

Edit the file to include:
+
[source,yaml]
----
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: aws-lb-operatorgroup
  namespace: aws-load-balancer-operator
spec:
  upgradeStrategy: Default
----

. Create the `subscription` file:

> [source,terminal]

Reviewer 3: ```suggestion
+
[source,terminal]
```

----
oc apply -f subscription.yaml
cat subscription.yaml
----

Edit the file to include:

[source,yaml]
----
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: aws-load-balancer-operator
  namespace: aws-load-balancer-operator
spec:
  channel: stable-v1
  installPlanApproval: Automatic
  name: aws-load-balancer-operator
  source: qe-app-registry
  sourceNamespace: openshift-marketplace
----

.Verification

Check the status of the install plan by running the following command:

> 

Reviewer 3: ```suggestion
+
```

[source,terminal]
----
oc -n aws-load-balancer-operator get sub aws-load-balancer-operator --template='{{.status.installplan.name}}{{"\n"}}'
oc -n aws-load-balancer-operator get ip <install_plan_name> --template='{{.status.phase}}{{"\n"}}'
----

> 

Reviewer 3: ```suggestion
+
```

// Module included in the following assemblies:
//
// * networking/aws_load_balancer_operator/install-aws-load-balancer-operator.adoc

:_mod-docs-content-type: PROCEDURE
[id="nw-installing-aws-load-balancer-operator-cli_{context}"]
= Installing the AWS Load Balancer Operator with the command line

.Prerequisites

* You have logged in to the {product-title} web console as a user with `cluster-admin` permissions.

* Your cluster is configured with AWS as the platform type and cloud provider.

.Procedure

> . Create aws-load-balancer-operator namespace: oc new-project aws-load-balancer-operator and the created OperatorGroup and Subscription:

Reviewer 3: ```suggestion
. Create the `aws-load-balancer-operator` namespace:
[source,terminal]
----
oc new-project aws-load-balancer-operator
----
```

// Module included in the following assemblies:
//
// * networking/aws_load_balancer_operator/install-aws-load-balancer-operator.adoc

:_mod-docs-content-type: PROCEDURE
[id="nw-installing-aws-load-balancer-operator-cli_{context}"]
= Installing the AWS Load Balancer Operator with the command line

.Prerequisites

> * You have logged in to the {product-title} web console as a user with `cluster-admin` permissions.

Reviewer 2: The user should be logged to the command line client: `oc`.

* Your cluster is configured with AWS as the platform type and cloud provider.

.Procedure

. Create the `aws-load-balancer-operator` namespace:

[source,terminal]
----
oc new-project aws-load-balancer-operator
----
+

> The following configurations will be automatically created:

Reviewer 2: Do we need to show what will be created before creating? Can't we just show the commands of how to create `OperatorGroup` and `Subscription`? Similar to how we created the namespace above.

Author: I think this question means the breakdown of steps is incorrect at the moment.

.Expected `OperatorGroup` object
+
[source,yaml]
----
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
name: aws-load-balancer-operator
namespace: aws-load-balancer-operator
spec:
upgradeStrategy: Default
----

.Expected `Subscription` object

[source,yaml]
----
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: aws-load-balancer-operator
  namespace: aws-load-balancer-operator
spec:
  channel: stable-v1.1
  installPlanApproval: Automatic
  name: aws-load-balancer-operator
  source: redhat-operators
  sourceNamespace: openshift-marketplace
----
. Apply the configurations by running the following command:
+
[source,terminal]
Suggested change
----
$ oc apply -f 
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
name: aws-load-balancer-operator
namespace: aws-load-balancer-operator
spec:

> upgradeStrategy: Default DO WE NEED THE FILENAME HERE

Reviewer 2: This can be replaced by an "in-place" command without any additional file:
```
cat <<EOF | oc apply -f -
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: aws-load-balancer-operator
  namespace: aws-load-balancer-operator
spec:
  targetNamespaces: []
EOF
```

Author: Per our standards, we do one step at a time. :)

Reviewer 2: Then you can split it in 2 commands:
```bash
cat <<EOF > filename.yaml
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: aws-load-balancer-operator
  namespace: aws-load-balancer-operator
spec:
  targetNamespaces: []
EOF
```

```bash
$ oc apply -f filename.yaml
```

Reviewer 4: @alebedev87 Per our style guidelines, we avoid `cat` commands. Is there a way to rework this to only use `oc` commands? Thanks!

Reviewer 2: Not really. I think we have to show the yaml and ask to save it into a file:

```yaml
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: aws-load-balancer-operator
  namespace: aws-load-balancer-operator
spec:
  targetNamespaces: []
```

The `oc apply -f <filename>` remains the same.

----
. Create a YAML file <Subscription> with the following configuration:
+
[source,yaml]
----
CODE HELP NEEDED
----
. Apply the configuration by running the following command:
+
[source,terminal]
----
$ oc apply -f <name_of_OperatorGroup_config>
----
+
[source,terminal]
----

> $ oc apply -f <name_of_Subscription_config>

Reviewer 2: Similar one line command can be used for the creation of `Subscription`:
```
cat <<EOF | oc apply -f -
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: aws-load-balancer-operator
  namespace: aws-load-balancer-operator
spec:
  channel: stable-v1
  name: aws-load-balancer-operator
  source: redhat-operators
  sourceNamespace: openshift-marketplace
EOF
```

Author: We like to break up commands into separate steps to be clear about what each step/line of code does, per our style guide.

Reviewer 2: You can do this then:

```bash
cat <<EOF > filename.yaml
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: aws-load-balancer-operator
  namespace: aws-load-balancer-operator
spec:
  channel: stable-v1
  name: aws-load-balancer-operator
  source: redhat-operators
  sourceNamespace: openshift-marketplace
EOF
```

```bash
oc apply -f filename.yaml
```

Reviewer 2: Here is the the yaml, in case it's the way to go:
```yaml
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: aws-load-balancer-operator
  namespace: aws-load-balancer-operator
spec:
  channel: stable-v1
  name: aws-load-balancer-operator
  source: redhat-operators
  sourceNamespace: openshift-marketplace
```
